### ISSUE TITLE
Cannot select InlineMath in Table

### ISSUE THREAD

**Issue by rriemann** [htdebeer/paru#37]
Hello,

I wont to write a filter to replace keywords with variables. Keywords use the same syntax like Pandoc template variables, i.e. `$keyword$`.

My `filter-replace.rb`:

```ruby
#!/usr/bin/env ruby
require 'paru/filter'

Paru::Filter.run do
  with "Math" do |str|
    STDERR.puts str.inner_markdown.inspect
  end
end
```

My markdown file for testing:

```markdown

# Title

In Paragraph: $replace.in_para$.

  In Table
  ---------
  $replace.in_table$
```

However, the selector filters only for the first key word in the paragraph, not the second in the table. How can I get both?

**Comment by htdebeer** [comment]
I can reproduce the issue. However, if I remove the indentation in front of the table it seems to be working, although the table header is interpreted as a Header.

If I export the markdown to json, I do see that in the indented case the Math is in a Plain in a Table, whereas in the non indented case the Math is in a Para in a Table.

It looks like paru does not like / understand Math in a Plain block. I'll investigate this issue.

**Comment by htdebeer** [comment]
Apparently, the issue was not with Plain nodes but with Table nodes: the rows of the table were not being filtered at all. 

I've created a fix; paru-0.3.0.1 should work

**Comment by rriemann** [comment]
Thank you a lot! I confirm this solves my problem.
